src/ecs/resources/input/input_handler.rs

```diff
@@ -36,8 +36,8 @@ pub struct InputHandler {
     pressed_mouse_buttons: SmallVec<[MouseButton; 16]>,
     down_mouse_buttons: SmallVec<[MouseButton; 16]>,
     released_mouse_buttons: SmallVec<[MouseButton; 16]>,
-    mouse_position: Option<(i32, i32)>,
-    previous_mouse_position: Option<(i32, i32)>,
+    mouse_position: Option<(f64, f64)>,
```

Comment:
Why do these need to change?  mouse positions on screen are integral values, so I don't see what value a float adds here, much less an f64 as those are more expensive in 32 bit architectures.  Many games are only released in 32 bit mode.  I see in winit these are f64 values now.  I still think we shouldn't use f64 values here, and we should instead cast the values from winit to i32.